tests/unit/externalRequestsTest.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const sinon = require('sinon')
+const tap = require('tap')
+
+const externalRequests = require('../../src/middleware/externalRequests')
+
+tap.test('External Requests Middleware', {autoend: true}, t => {
+  t.test('prepareLookupRequests', {autoend: true}, t => {
+    let performRequestsStub
+    t.beforeEach(done => {
+      performRequestsStub = sinon.stub(externalRequests, 'performRequests')
+      done()
+    })
+
+    t.afterEach(done => {
+      performRequestsStub.restore()
+      done()
+    })
+
+    t.test('should throw an error if any promise in the array fails', t => {
+      t.plan(1)
+      performRequestsStub
+        .onFirstCall()
+        .returns([Promise.resolve('Success'), Promise.reject('Fail')])
+
+      const ctx = {
+        state: {
+          uuid: 'randomUidForRequest',
+          metaData: {
+            name: 'Testing endpoint',
+            requests: {
+              lookup: [{}, {}]
+            }
+          }
+        }
+      }
+
+      externalRequests
+        .prepareLookupRequests(ctx)
+        .then(() => {
+          t.error('Should not reach here')
+        })
+        .catch(err => {
+          t.throws(err, 'Error should be thrown when a promise rejects')
```

Comment:
lets add in another assert to verify the error being thrown is the one we expected. That the lookup failed and not some other internal error

Reply:
Done in https://github.com/jembi/openhim-mediator-mapping/pull/37/commits/cd7944971d30fc94d7136237ac5364e066c61fb3